.github/workflows/performance-profiling.yml

```diff
@@ -4,6 +4,10 @@ on:
   pull_request:
     types: [opened, synchronize, ready_for_review]
   workflow_dispatch:
+concurrency:
```

Comment:
Can you please also add a filter on which files trigger this workflow? Only files in `src/zenml` should be relevant I assume?